### ISSUE TITLE
Documentation for objective functions R package

### ISSUE THREAD

**Issue by kransom14** [dmlc/xgboost#7445]
Where can I find documentation for how the different objective functions are calculated? For example, "reg:tweedie". What is the equation for the loss function as well as the gradient and the hessian?

**Comment by trivialfis** [comment]
Unfortunately, the related documents are sparse.  I will try to put a brief description in the doc.
